test/unit/client_test.coffee

```diff
@@ -107,6 +108,12 @@ describe "Client", ->
         error = _err
       return error
 
+    it "returns promise and resolves it", ->
+      promise = client.push(exception)
```

Comment:
> callbacks are considered a bad practise nowadays

Well, *overusing* callbacks is bad practise (always was). I'd argue they work just fine for simple cases though.

> Promise API is going to be a part of ES6 spec.

But the code in this PR isn't just a polyfill that is loaded only for old browser, it seems like a (partial?) implementation of ES6 Promise. I think that is something we should avoid if possible. Why not use an existing polyfill (maybe https://github.com/jakearchibald/es6-promise)?